### ISSUE TITLE
질문

### ISSUE THREAD

**Issue by moonyoungCHAE** [public-ai/ALAI_Homework#51]
1. Boston House  결과
![loss_func](https://user-images.githubusercontent.com/37133536/58536657-2c9a6a80-822c-11e9-8c7d-6f0094e09748.PNG)
2 Layer로 구현했을 때, 이렇게 나오는 데 정상적인 것인가요? loss 가 더 떨어져야 하는 것인지 궁금합니다.

2. Tensorflow 문법
모멘텀을 구현하고 있는데
~~~
def rosenbrock (x, y):    
    return (1-x) ** 2 + 100*(y - x**2) ** 2

start_x, start_y = -4., -4.
learning_rate = 0.01
max_iter = 10

x = tf.Variable(initial_value=tf.constant(start_x), name='x')
y = tf.Variable(initial_value=tf.constant(start_y), name='y')
z = rosenbrock(x, y)
z = tf.identity(z, name = "z")
lr = tf.Variable(initial_value=tf.constant(learning_rate), name='lr')

ass_x = tf.assign(x, x - tf.math.multiply(lr, tf.gradients(ys = z, xs = x)))
ass_y = tf.assign(y, y - tf.math.multiply(lr, tf.gradients(ys = z, xs = y)))

sess = tf.Session()
sess.run(tf.global_variables_initializer())

gd_xs = []
gd_ys = []
gd_zs = []
for i in range(max_iter):
    x_, y_, z_ = sess.run([x, y, z])
    gd_xs.append(x_)
    gd_ys.append(y_)
    gd_zs.append(z_)
    
    sess.run([ass_x, ass_y])
~~~

여기서 x의 shape는 () 고 tf.gradients 후에는 (1,) 이렇게 나와서 shape가 안 맞는데요.
z, x의 차원을 확대한 후 tr.gradients하면 None이 나오고,
assign할 때 x의 차원을 확대하면 'Tensor' object has no attribute 'assign' 에러가 나는 데 어떻게 해결하나요?

**Comment by anthony0727** [comment]
1. loss가 잘 떨어지는 것 같습니다!

2. `tf.gradients(ys = z, xs = x)[0]`으로 수정해주시면 될 것 같습니다. tf.gradients는 list를 반환합니다!
